action.yaml

```diff
@@ -106,7 +106,19 @@ runs:
         fi
         echo "dependency_list=${dependencies}" >> $GITHUB_OUTPUT
 
-        dependencies_sha=$(echo ${dependencies} | md5sum)
+        deps_w_versions=""
+        sudo apt update
+        for dep in ${dependencies}
+        do
+          # Simulate package install to fetch package version;
+          # head -n1 in case there are multiple packages being installed (eg: for qemu-system);
+          # Output line similar to:
+          # Inst virtinst (1:4.1.0-3ubuntu0.1 Ubuntu:24.04/noble-updates [all])
+          version=$(apt-get -s install $dep | grep "Inst $dep"  | head -n1 | awk -F" " '{ printf $3 }' | cut -c2-)
+          deps_w_versions="${dep}_${version} ${deps_w_versions}"
+        done
+        # Remove ending "-" character
```

Comment:
do you know why the - character was here in the first place? I'm not sure to see from the example
```
Inst virtinst (1:4.1.0-3ubuntu0.1 Ubuntu:24.04/noble-updates [all])
```

Reply:
It comes from `md5sum` output:
```
$ echo "test" | md5sum
d8e8fca2dc0f896fd7cb4cb0031ba249  -
```

Reply:
Clarified in the comment :)

Reply:
aaaah indeed! thanks